addons/flannel/0.21.5/install.sh

```diff
@@ -316,11 +316,25 @@ function remove_weave() {
     kubectl -n kube-system delete serviceaccount weave-net
     kubectl -n kube-system delete secret weave-passwd
 
-    # all nodes
-    ip link delete weave
+    # It should be executed in all nodes either
+    # See that the task used to run in the nodes also has this commands
+    # Delete the weave network interface, if it exists
+    if ip link show weave > /dev/null 2>&1; then
+        ip link delete weave
+    fi
+
+    # Delete the /var/lib/weave directory, if it exists
     rm -rf /var/lib/weave
-    rm -rf /etc/cni/net.d/*weave*
-    rm -rf /opt/cni/bin/weave*
+
+    # Delete any weave files in /etc/cni/net.d
+    if ls /etc/cni/net.d/*weave* > /dev/null 2>&1; then
+        rm -rf /etc/cni/net.d/*weave*
+    fi
```

Comment:
this is not necessary. the -f argument takes care of not found

```suggestion
    rm -rf /etc/cni/net.d/*weave*
```